xds/src/test/java/io/grpc/xds/GcpAuthenticationFilterTest.java

```diff
@@ -0,0 +1,121 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.Message;
+import com.google.protobuf.UInt64Value;
+import io.envoyproxy.envoy.extensions.filters.http.gcp_authn.v3.GcpAuthnFilterConfig;
+import io.envoyproxy.envoy.extensions.filters.http.gcp_authn.v3.TokenCacheConfig;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientInterceptor;
+import io.grpc.MethodDescriptor;
+import io.grpc.testing.TestMethodDescriptors;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+@RunWith(JUnit4.class)
+public class GcpAuthenticationFilterTest {
+
+  @Test
+  public void testParseFilterConfig_withValidConfig() {
+    GcpAuthnFilterConfig config = GcpAuthnFilterConfig.newBuilder()
+        .setCacheConfig(TokenCacheConfig.newBuilder().setCacheSize(UInt64Value.of(10)))
+        .build();
+    Any anyMessage = Any.pack(config);
+
+    GcpAuthenticationFilter filter = new GcpAuthenticationFilter();
+    ConfigOrError<? extends Filter.FilterConfig> result = filter.parseFilterConfig(anyMessage);
+
+    assertNotNull(result.config);
+    assertNull(result.errorDetail);
+    assertEquals(10L,
+        ((GcpAuthenticationFilter.GcpAuthenticationConfig)
+            result.config).getCacheSize());
+  }
+
+  @Test
+  public void testParseFilterConfig_withZeroCacheSize() {
+    GcpAuthnFilterConfig config = GcpAuthnFilterConfig.newBuilder()
+        .setCacheConfig(TokenCacheConfig.newBuilder().setCacheSize(UInt64Value.of(0)))
+        .build();
+    Any anyMessage = Any.pack(config);
+
+    GcpAuthenticationFilter filter = new GcpAuthenticationFilter();
+    ConfigOrError<? extends Filter.FilterConfig> result = filter.parseFilterConfig(anyMessage);
+
+    assertNull(result.config);
+    assertNotNull(result.errorDetail);
+    assertTrue(result.errorDetail.contains("cache_config.cache_size must be in the range"));
+  }
+
+  @Test
+  public void testParseFilterConfig_withInvalidMessageType() {
+    GcpAuthenticationFilter filter = new GcpAuthenticationFilter();
+    Message invalidMessage = Mockito.mock(Message.class);
+    ConfigOrError<? extends Filter.FilterConfig> result = filter.parseFilterConfig(invalidMessage);
+
+    assertNull(result.config);
+    assertNotNull(result.errorDetail);
+    assertTrue(result.errorDetail.contains("Invalid config type"));
```

Comment:
`assertTrue()` produces particularly unhelpful error messages. Prefer Truth. `assertThat(result.errorDetail).contains("Invalid config type")` produces much better results. You also then don't need the `assertNotNull()` check.